language-design.md

```diff
@@ -17,7 +17,20 @@ For each quote, describe what it was about the quote that led you pick it.
 
 **Response**
 
+***Quote 1***
+>"I now think that I, as a language designer who helps out with the design of the Java programming language, need to ask not 'Should the Java programming language grow?’ but ‘How should the Java programming language grow?'" [Steele, 1998, 7]
 
+When boiled down to its very essence, this quote encapsulates the overall philosophy of this paper as well as Steele’s advice to all language designers. Specifically, this quote talks of the importance of language growth and evolution. A programming language does not exist to be a rigid set of unchanging rules. Rather, it exists as a tool for enabling powerful communication between humans and machines. Therefore, if there is a way to better achieve this communication, the language needs to adapt to the needs of its users and enable maximal ease and expressiveness. For these reasons, language evolution is not optional, but a necessity that motivates Steele’s claim that the “…main goal in designing a language should be to plan for growth” [Steele, 1998, 7]. Although Steele goes on to explain ideal methodologies for growing a language, citing the Linux development model as an exemplary case, the principal goal comes back to prioritizing the needs of the users. It is for this reason that Steele advocates creating languages as a more accessible platform that enables extensibility and natural evolution through things like libraries and extendable syntax.
+
+***Quote 2***
+>"We should not make the Java programming language a cathedral, but a plain bazaar might be too loose. What we need is more like a shopping mall, where there are not quite as many choices..." [Steele, 1998, 12]
+
+One of the challenges of language design is properly allowing room for users to extend and change the language. In this quote, Steele references and extends a metaphor that identifies the middle ground between no opportunity for user extension and too much wiggle room to provide cohesion in a language. While he talked about this idea at greater length that what we quoted, we felt that this summed up the key points. Essentially, a cathedral represents a language whose extension is inextensible by its user base. It is far too sturdy a structure and is too imposing to allow for creativity and change. Conversely, the bazaar represents the confusion of too little structure. While it has all the mobility and room for change one might desire, it does so at the cost of unity and clarity. As such, a shopping mall may be the happy middle ground. It has distinct structure as a building but allows for interchangeable parts and development.
```

Comment:
The bazaar/cathedral thing was a pretty interesting metaphor, and I agree with you that a middle ground is a good place to go here.